### PR TITLE
Provide dev Docker image for easier testing

### PR DESCRIPTION
**Pull request by reinkrul**
Fixes https://github.com/nuts-foundation/nuts-node/issues/2866

TODO:
- [x] Handle SIGKILL, ^C does not seem to work any more now
- [x] Re-use same localtunnel domain, if possible
- [ ] Documentation

### CODE REVIEW

**Comment by reinkrul** [comment]
Use of `exec` in entrypoint script causes signals to be captured by Nuts node, causing `^C` to be handled. But now it looks like the process is killed before the process completes its shutdown;

```
time="2024-03-11T08:35:29Z" level=info msg="Shutting down..."
time="2024-03-11T08:35:29Z" level=info msg="Stopping HTTP..." module=core
```

**Comment by reinkrul** [comment]
Documentation should be in "Running on Docker", but that's partly rewritten by https://github.com/nuts-foundation/nuts-node/pull/2884

Waiting for that PR.

File: development/dev-image/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  cat /localtunnel.log
+  exit 1
+fi
+# Extract subdomain (TUNNEL_URL after https:// up to the first .)
+SUBDOMAIN=$(echo $TUNNEL_URL | awk -F/ '{print $3}' | awk -F. '{print $1}')
+# Store subdomain in localtunnel.cfg so that the container retains its subdomain after restart
+echo "${SUBDOMAIN}" > /localtunnel.cfg
+
+echo Your Nuts node URL is: ${TUNNEL_URL}
+NUTS_URL="${TUNNEL_URL}" NUTS_STRICTMODE=false NUTS_AUTH_CONTRACTVALIDATORS=dummy exec /usr/bin/nuts server
```
Comment (woutslakhorst) [inline development/dev-image/entrypoint.sh:32]: The persistent hostname is not working for me, here is an example that uses the `TUNNEL_NAME` env var. It also kills both processes.

```suggestion
#!/bin/sh

# Try to use TUNNEL_NAME as subdomain
SUBDOMAIN=${TUNNEL_NAME}
echo Trying $SUBDOMAIN as subdomain

# Run lt to create the local tunnel with --port 8080 as background process.
# It writes the URL to stdout which we need to extract and feed to the nuts command
# capture the pid in a file so we can kill the process when the container stops
lt --port 8080 --subdomain="${SUBDOMAIN}" > /localtunnel.log 2>&1 &
echo $! > /localtunnel.pid

# Try 30 times to read the tunnel URL from the log file.
# The format is "your url is:" followed by the URL.
echo "Waiting for localtunnel URL (time-out in 30s)..."
for i in $(seq 1 30); do
  TUNNEL_URL=$(grep "your url is:" /localtunnel.log | awk '{print $4}')
  if [ -n "$TUNNEL_URL" ]; then
    break
  fi
  sleep 1
done
# Check whether we retrieved the URL and if not, exit with an error.
if [ -z "$TUNNEL_URL" ]; then
  echo "Failed to retrieve the localtunnel URL"
  cat /localtunnel.log
  exit 1
fi
cat /localtunnel.log
# Extract subdomain (TUNNEL_URL after https:// up to the first .)
SUBDOMAIN=$(echo $TUNNEL_URL | awk -F/ '{print $3}' | awk -F. '{print $1}')

echo Your Nuts node URL is: ${TUNNEL_URL}
NUTS_URL="${TUNNEL_URL}" NUTS_STRICTMODE=false NUTS_AUTH_CONTRACTVALIDATORS=dummy exec /usr/bin/nuts server

# Kill the localtunnel process when the container stops
kill $(cat /localtunnel.pid)
```

**Comment by reinkrul** [comment]
Replaced by https://github.com/nuts-foundation/nuts-node/pull/2915

**Pull request closed without merging**